cmd/activator/handlers.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2018 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+
+	"github.com/knative/serving/pkg/activator"
+	"github.com/knative/serving/pkg/controller"
+	"go.uber.org/zap"
+)
+
+// activationHandler will proxy a request to the active endpoint for the specified revision,
+// using the provided transport
+type activationHandler struct {
+	activator activator.Activator
+	logger    *zap.SugaredLogger
+	transport http.RoundTripper
+}
+
+func newActivationHandler(a activator.Activator, rt http.RoundTripper, l *zap.SugaredLogger) http.Handler {
+	return &activationHandler{activator: a, transport: rt, logger: l}
+}
+
+func (a *activationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	namespace := r.Header.Get(controller.GetRevisionHeaderNamespace())
+	name := r.Header.Get(controller.GetRevisionHeaderName())
+
+	endpoint, status, err := a.activator.ActiveEndpoint(namespace, name)
+	if err != nil {
+		msg := fmt.Sprintf("Error getting active endpoint: %v", err)
+
+		a.logger.Errorf(msg)
+		http.Error(w, msg, int(status))
+
+		return
+	}
+
+	target := &url.URL{
+		Scheme: "http",
+		Host:   fmt.Sprintf("%s:%d", endpoint.FQDN, endpoint.Port),
+	}
+
+	proxy := httputil.NewSingleHostReverseProxy(target)
+	proxy.Transport = a.transport
+
+	// TODO: Clear the host to avoid 404's.
+	// https://github.com/knative/serving/issues/964
+	r.Host = ""
+
+	proxy.ServeHTTP(w, r)
+}
+
+// uploadHandler wraps the provided handler with a request body that supports
+// re-reading and prevents uploads larger than `maxUploadBytes`
+type uploadHandler struct {
+	http.Handler
+	MaxUploadBytes int64
+}
+
+func newUploadHandler(h http.Handler, maxUploadBytes int64) http.Handler {
+	return uploadHandler{
+		Handler:        h,
+		MaxUploadBytes: maxUploadBytes,
+	}
+}
+
+func (h uploadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
Can you define the method on the pointer rather than the value type?
`func (h *uploadHandler) ServeHTTP(...)`